### PR TITLE
Fix build on FreeBSD

### PR DESCRIPTION
**Pull request by swills**
This fixes build on FreeBSD

### CODE REVIEW

**Comment by gdamore** [comment]
Why is this busted on FreeBSD?  I thought FreeBSD had these SOL_LOCAL socket options?

I'd rather fix this properly if possible.  And move the tests to the coverage.

File: src/platform/posix/posix_ipcconn.c [hunk 1]
```diff
@@ -363,7 +363,7 @@ ipc_peerid(ipc_conn *c, uint64_t *euid, uint64_t *egid, uint64_t *prid,
 	*prid = uc.pid;
 	*znid = (uint64_t) -1;
 	return (0);
-#elif defined(NNG_HAVE_LOCALPEERCRED)
+#elif defined(NNG_HAVE_LOCALPEERCRED) && !defined(__FreeBSD__)
```
Comment (gdamore) [inline src/platform/posix/posix_ipcconn.c:366]: FreeBSD does some to have a libc getpeereid() call that might be superior here anyway.

**Comment by swills** [comment]
Sorry, I was just after the quick fix. Here's the error I saw:

```
[  1%] Building C object src/CMakeFiles/nng_testlib.dir/platform/posix/posix_ipcconn.c.o
cd /usr/home/swills/Documents/git/nng_build/src && /usr/bin/cc -DNNG_ENABLE_STATS -DNNG_HAVE_ARC4RANDOM=1 -DNNG_HAVE_BUS0 -DNNG_HAVE_CLOCK_GETTIME=1 -DNNG_HAVE_FLOCK=1 -DNNG_HAVE_GETPEEREID=1 -DNNG_HAVE_GETRANDOM=1 -DNNG_HAVE_KQUEUE=1 -DNNG_HAVE_LOCALPEERCRED=1 -DNNG_HAVE_LOCKF=1 -DNNG_HAVE_MSG_CONTROL=1 -DNNG_HAVE_PAIR0 -DNNG_HAVE_PAIR1 -DNNG_HAVE_PTHREAD_ATFORK_PTHREAD=1 -DNNG_HAVE_PUB0 -DNNG_HAVE_PULL0 -DNNG_HAVE_PUSH0 -DNNG_HAVE_REP0 -DNNG_HAVE_REQ0 -DNNG_HAVE_RESPONDENT0 -DNNG_HAVE_SEMAPHORE_PTHREAD=1 -DNNG_HAVE_STDATOMIC=1 -DNNG_HAVE_STRCASECMP=1 -DNNG_HAVE_STRLCPY=1 -DNNG_HAVE_STRNCASECMP=1 -DNNG_HAVE_STRNLEN=1 -DNNG_HAVE_SUB0 -DNNG_HAVE_SURVEYOR0 -DNNG_HAVE_UNIX_SOCKETS=1 -DNNG_HIDDEN_VISIBILITY -DNNG_LITTLE_ENDIAN -DNNG_MAX_TASKQ_THREADS=16 -DNNG_PLATFORM_FREEBSD -DNNG_PLATFORM_POSIX -DNNG_PRIVATE -DNNG_STATIC_LIB -DNNG_SUPP_HTTP -DNNG_TEST_LIB -DNNG_TRANSPORT_INPROC -DNNG_TRANSPORT_IPC -DNNG_TRANSPORT_TCP -DNNG_TRANSPORT_TLS -DNNG_TRANSPORT_WS -D_GNU_SOURCE -D_POSIX_PTHREAD_SEMANTICS -D_REENTRANT -D_THREAD_SAFE -I/usr/home/swills/Documents/git/nng/include -isystem /usr/home/swills/Documents/git/nng/src -Wall -Wextra -fno-omit-frame-pointer     -std=gnu99 -o CMakeFiles/nng_testlib.dir/platform/posix/posix_ipcconn.c.o   -c /usr/home/swills/Documents/git/nng/src/platform/posix/posix_ipcconn.c
/usr/home/swills/Documents/git/nng/src/platform/posix/posix_ipcconn.c:369:21: error: use of undeclared identifier 'SOL_LOCAL'
        if (getsockopt(fd, SOL_LOCAL, LOCAL_PEERCRED, &xu, &len) != 0) {
                           ^
1 error generated.
*** Error code 1


```

The `sys/un.h` file doesn't seem to have `SOL_LOCAL`, see [sys/un.h](https://github.com/freebsd/freebsd/blob/master/sys/sys/un.h)

**Comment by swills** [comment]
FWIW, [getpeereid()](https://github.com/freebsd/freebsd/blob/master/lib/libc/gen/getpeereid.c#L50) just has zero for the second (level) arg to getsockopt(), so probably just defining SOL_LOCAL to 0 is fine.

**Comment by swills** [comment]
FWIW, the `SOL_LOCAL` define was added:

https://github.com/freebsd/freebsd/commit/bed966847f40df2faad9b99a8a28730782e2f191

**Comment by gdamore** [comment]
Wow.  That's funny.  I think we could probably for now just change the code to do #ifndef SOL_LOCAL ... #define .... #endif ...

**Comment by swills** [comment]
Hmm, is that sanitize failure legit or some intermittent or unrelated issue?

File: src/platform/posix/posix_ipcconn.c [hunk 2]
```diff
@@ -33,6 +33,12 @@
 #define MSG_NOSIGNAL 0
 #endif
 
+#ifndef SOL_LOCAL
+#ifdef __FreeBSD__
```
Comment (gdamore) [inline src/platform/posix/posix_ipcconn.c:37]: I don't think you need the __FreeBSD__ guard here.  I'm fairly confident that the systems in question either already have it defined, or use 0 (and not some other value.)

**Comment by gdamore** [comment]
Looks good, I'll merge once the CI run is complete.

**Pull request merged**